File: js/189_rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function (nums, k) {

    size = nums.length
    temp = [...nums]

    for (let i = 0; i < size; i++) {
        nums[(i + k) % size] = temp[i]
    }

};

File: python/189_rotate-array.py
from typing import List

class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        size = len(nums)
        temp = nums.copy()

        for i in range(size):
            nums[(i+k) % size] = temp[i]

# Time:  O(n)
# Space: O(n)
